src/main/kotlin/lotto/domain/LottoMatcher.kt

```diff
@@ -1,37 +1,27 @@
 package lotto.domain
 
-@JvmInline value class EarnedRate(val rate: Float)
-@JvmInline value class EarnedMoney(val money: Long)
-
-data class LottoMatchResult(
-    val matchResult: Map<WinningInfo, Int>,
-    val earnedMoney: EarnedMoney,
-)
-
 class LottoMatcher() {
-    fun matchResult(lottoTickets: LottoTickets, winningNumbers: WinningNumber): LottoMatchResult {
-        val matchedResult = getMatchedResult(lottoTickets, winningNumbers)
-        val earnedMoney = getEarnedMoney(matchedResult)
-        return LottoMatchResult(matchedResult, earnedMoney)
+    fun matchResult(lottoTickets: LottoTickets, winningNumbers: WinningLotto, bonusNumber: BonusNumber): LottoMatchResult {
+        val matchResult = formatMatchResult(
+            lottoTickets.match(winningNumbers, bonusNumber)
+        )
+        val earnedMoney = getEarnedMoney(matchResult)
+        return LottoMatchResult(matchResult, earnedMoney)
     }
 
     fun calculateEarnedRate(earnedMoney: EarnedMoney, paidMoney: Long): EarnedRate {
         val earnedRate = earnedMoney.money / paidMoney.toFloat()
         return EarnedRate(earnedRate)
     }
 
-    private fun getMatchedResult(lottoTickets: LottoTickets, winningNumbers: WinningNumber): Map<WinningInfo, Int> {
-        return generateMatchResult(lottoTickets.match(winningNumbers))
-    }
-
-    private fun generateMatchResult(winningCountMap: Map<WinningInfo, Int>): Map<WinningInfo, Int> {
-        return WinningInfo.values().associateWith { winningInfo -> winningCountMap.getOrDefault(winningInfo, 0) }
-    }
+    private fun formatMatchResult(resultMap: Map<Rank, Count>) =
+        Rank.values().associateWith { resultMap.getOrDefault(it, Count(0)) }
 
-    private fun getEarnedMoney(matchedResult: Map<WinningInfo, Int>): EarnedMoney {
-        val earnedMoney = matchedResult
-            .map { (winningInfo, winningCount) -> winningInfo.winningMoney * winningCount }
-            .sum()
-        return EarnedMoney(earnedMoney)
+    private fun getEarnedMoney(matchResult: Map<Rank, Count>): EarnedMoney {
+        return EarnedMoney(
+            matchResult
+                .map { (rank, count) -> rank.winningMoney * count.count }
+                .sum()
+        )
```

Comment:
위에 코드와 같이 일관성 있게 변경할 수 있어보여요
```suggestion
    private fun getEarnedMoney(matchResult: Map<Rank, Count>): EarnedMoney =
        EarnedMoney(
            matchResult
                .map { (rank, count) -> rank.winningMoney * count.count }
                .sum()
        )
```